cpp/core.h

```diff
@@ -55,7 +55,7 @@ class TermState {
   }
 };
 
-class SignalHandler {
+class SignalHandler : public Obj {
```

Comment:
Without this we lose track of the new signal queue and get a use-after-free

Reply:
OK interesting, yeah I can see why having a global pointing to a GC managed object is probelmatic

But don't you think we need to also root the global?